eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelConfigOptions.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.channel
+
+/**
+ * Created by t-bast on 24/06/2021.
+ */
+
+/**
+ * Internal configuration option impacting the channel's structure or behavior.
+ * This must be set when creating the channel and cannot be changed afterwards.
+ */
+trait ChannelConfigOption {
```

Comment:
Why? It's really a configuration option...it's not a complete configuration so `ChannelConfig` feels weird IMO, and `ChannelOption` is on the contrary a bit too vague, isn't it?

Reply:
It's just that everywhere we are using: `channelConfig: ChannelConfigOptions`

Reply:
But it's different, it's because it's a `ChannelConfigOptions`, which is a collection of `ChannelConfigOption`.
I agree this one could be renamed `ChannelConfigOptions` -> `ChannelConfig`, but not the trait on individual options.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1852/commits/4e0c81413694e47662e5a21ed4d8a53d19d5e983